ast/asdl_rs.py

```diff
@@ -332,6 +327,7 @@ def sum_subtype_struct(self, sum_type_info, t, rust_name, depth):
         self.emit(f"pub struct {payload_name}{generics} {{", depth)
         for f in t.fields:
             self.visit(f, sum_type_info, "pub ", depth + 1, t.name)
+        self.emit("pub range: TextRange", depth + 1)
```

Comment:
I recently realized range is a sort of custom.

Currently RustPython uses ast with both location as custom and and textrange as hard-coded type. 

```suggestion
        self.emit("pub custom: U", depth + 1)
```
By changing this field, `Ranged` trait can work just same as before and it can be replaced out when it is not used.

Reply:
Ah, this is the variant type generation. Isn't it expected to be added to `sum_with_constructors`? Like `Stmt` and `Expr` types.

Reply:
You mean, ruff needs to add range field to literally every nodes even if a few of the original Python ast nodes doesn't have location information?

Reply:
Yes